plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -226,23 +226,41 @@ def process_main(self):
         # only want to download each layer once.
         available_blobs = set()
         self.total_units = len(available_tags)
+        upstream_name = self.config.get('upstream_name')
         for tag in available_tags:
-            digest, manifest = self.parent.index_repository.get_manifest(tag)
-            # Save the manifest to the working directory
-            with open(os.path.join(self.get_working_dir(), digest), 'w') as manifest_file:
-                manifest_file.write(manifest)
-            manifest = models.Manifest.from_json(manifest, digest)
-            self.parent.available_manifests.append(manifest)
-            for layer in manifest.fs_layers:
-                available_blobs.add(layer.blob_sum)
-            # Remember this tag for the SaveTagsStep.
-            self.parent.save_tags_step.tagged_manifests[tag] = manifest
+            # to understand how the manifests list's indexes are used look at get_manifest()
+            manifests = self.parent.index_repository.get_manifest(tag)
```

Comment:
i would like to leave the return value of get_manifest() as it is, but i will make some changes here, so it would nit depend on the manifests return structure